cli/main.go

```diff
@@ -28,6 +37,98 @@ func main() {
 		EnableBashCompletion:   true,
 		UseShortOptionHandling: true,
 		Commands: []*cli.Command{
+			{
+				Name:  "status",
+				Usage: "Checks the status of your eigenpod.",
+				Flags: []cli.Flag{
+					&cli.BoolFlag{
+						Name:        "json",
+						Value:       false,
+						Usage:       "print only plain JSON",
+						Required:    false,
+						Destination: &useJson,
+					},
+				},
+				Action: func(cctx *cli.Context) error {
+					eth, err := ethclient.Dial(node)
+					PanicOnError("failed to reach eth --node.", err)
+
+					beaconClient, err := getBeaconClient(beacon)
+					PanicOnError("failed to reach beacon chain.", err)
+
+					status := getStatus(ctx, eigenpodAddress, eth, beaconClient)
+					eigenpod, err := onchain.NewEigenPod(common.HexToAddress(eigenpodAddress), eth)
+					PanicOnError("failed to reach eigenpod", err)
+
+					if useJson {
+						bytes, err := json.MarshalIndent(status, "", "      ")
+						PanicOnError("failed to get status", err)
+						statusStr := string(bytes)
+						fmt.Println(statusStr)
+					} else {
+						// pretty print everything
+						color.New(color.Bold, color.FgBlack).Printf("Eigenpod validators\n")
+						for index, validator := range status.Validators {
+
+							var targetColor color.Attribute
+							var description string
+
+							if validator.Status == ValidatorStatusActive {
+								targetColor = color.FgGreen
+								description = "active"
+							} else if validator.Status == ValidatorStatusInactive {
+								targetColor = color.FgHiYellow
+								description = "inactive"
+							} else if validator.Status == ValidatorStatusWithdrawn {
+								targetColor = color.FgHiRed
+								description = "withdrawn"
+							}
+
+							color.New(targetColor).Printf("\t- #%s (%s) [%s]\n", index, shortenAddress(validator.PublicKey), description)
+						}
+
+						bold := color.New(color.Bold, color.FgBlack)
+						ital := color.New(color.Italic, color.FgBlack)
+						fmt.Println()
+
+						eigenpodManagerContractAddress, err := eigenpod.EigenPodManager(nil)
+						PanicOnError("failed to get manager address", err)
+
+						eigenPodManager, err := onchain.NewEigenPodManager(eigenpodManagerContractAddress, eth)
+						PanicOnError("failed to get manager instance", err)
+
+						eigenPodOwner, err := eigenPodManager.Owner(nil)
+						PanicOnError("failed to get eigenpod owner", err)
+
+						ownerShares, err := eigenPodManager.PodOwnerShares(nil, eigenPodOwner)
```

Comment:
This is a signed integer - and i think the conversion to Float64 below doesn't factor that in

Reply:
yeah i'm gonna gut the math in this and convert everything to bigInts/bigFloats today